packages/dx-react-scheduler-demos/src/demo-sources/scheduler-basic/material-ui/custom-template.jsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import Paper from '@material-ui/core/Paper';
+import { ViewState } from '@devexpress/dx-react-scheduler';
+import {
+  Scheduler,
+  WeekView,
+  Appointments,
+} from '@devexpress/dx-react-scheduler-material-ui';
+import { withStyles } from '@material-ui/core/styles';
+import { fade } from '@material-ui/core/styles/colorManipulator';
+import appointments from '../../../demo-data/today-appointments';
+
+const style = theme => ({
+  todayCell: {
+    backgroundColor: fade(theme.palette.primary.main, 0.1),
+    '&:hover': {
+      backgroundColor: fade(theme.palette.primary.main, 0.14),
+    },
+    '&:focus': {
+      backgroundColor: fade(theme.palette.primary.main, 0.16),
+    },
+  },
+  weekendCell: {
+    backgroundColor: fade(theme.palette.action.disabledBackground, 0.04),
+    '&:hover': {
+      backgroundColor: fade(theme.palette.action.disabledBackground, 0.04),
+    },
+    '&:focus': {
+      backgroundColor: fade(theme.palette.action.disabledBackground, 0.04),
+    },
+  },
+  today: {
+    backgroundColor: fade(theme.palette.primary.main, 0.16),
+  },
+  weekend: {
+    backgroundColor: fade(theme.palette.action.disabledBackground, 0.06),
+  },
+});
+
+const TimeTableCellBase = ({ classes, ...restProps }) => {
+  const { startDate } = restProps;
+  const date = new Date(startDate);
+  if (date.getDate() === new Date().getDate()) {
+    return <WeekView.TimeTableCell {...restProps} className={classes.todayCell} />;
+  } if (date.getDay() === 0 || date.getDay() === 6) {
```

Comment:
Isn't *newline* lost here?

Reply:
Where? All are fine in my opinion.

Reply:
What case do we have here?

```javascript
if (condition1) {
   ...
} else if (condition2) {
  ...
}
```

```javascript
if (condition1) {
  ...
}
if (condition2) {
 ...
}
```

Reply:
I mean `else if` pattern because we use `return` here

Reply:
Then `else` should present there - it looks weird otherwise.
However - it it considered normal then keep it.